### DISCUSSION TITLE
Is it possible to install php73-intl on the ubuntu-latest OS?

### DISCUSSION THREAD

**Discussion by DeBelserArne** [community/community#26210] [Actions]
<p>I’m running the following bit of code in my .yml</p>
<pre><code>- name: Install ext-intl
      run: sudo apt-get install php7.3-intl
</code></pre>
<p>But this doesn’t really work, here’s the log:</p>
<p></p><div><div><span><span>Preparing to unpack …/php7.3-intl_7.3.11-1+ubuntu18.04.1+deb.sury.org+1_amd64.deb … </span></span></div></div><p></p>
<div><div>
<a href="https://github.com/aFluxx/Pigero/commit/b949dfd06c0cb21456a7c04b0ad6c3406d7189f4/checks?check_suite_id=292928151#step:5:56" target="_blank" rel="nofollow noopener">56</a><span><span>Unpacking php7.3-intl (7.3.11-1+ubuntu18.04.1+deb.sury.org+1) ... </span></span>
</div></div>
<div><div>
<a href="https://github.com/aFluxx/Pigero/commit/b949dfd06c0cb21456a7c04b0ad6c3406d7189f4/checks?check_suite_id=292928151#step:5:57" target="_blank" rel="nofollow noopener">57</a><span><span>Setting up libicu64:amd64 (64.2-0.1+ubuntu18.04.1+deb.sury.org+1) ... </span></span>
</div></div>
<div><div>
<a href="https://github.com/aFluxx/Pigero/commit/b949dfd06c0cb21456a7c04b0ad6c3406d7189f4/checks?check_suite_id=292928151#step:5:58" target="_blank" rel="nofollow noopener">58</a><span><span>Setting up php7.3-intl (7.3.11-1+ubuntu18.04.1+deb.sury.org+1) ... </span></span>
</div></div>
<div><div>
<a href="https://github.com/aFluxx/Pigero/commit/b949dfd06c0cb21456a7c04b0ad6c3406d7189f4/checks?check_suite_id=292928151#step:5:59" target="_blank" rel="nofollow noopener">59</a><span><span>debconf: unable to initialize frontend: Dialog </span></span>
</div></div>
<div><div>
<a href="https://github.com/aFluxx/Pigero/commit/b949dfd06c0cb21456a7c04b0ad6c3406d7189f4/checks?check_suite_id=292928151#step:5:60" target="_blank" rel="nofollow noopener">60</a><span><span>debconf: (Dialog frontend will not work on a dumb terminal, an emacs shell buffer, or without a controlling terminal.) </span></span>
</div></div>
<div><div>
<a href="https://github.com/aFluxx/Pigero/commit/b949dfd06c0cb21456a7c04b0ad6c3406d7189f4/checks?check_suite_id=292928151#step:5:61" target="_blank" rel="nofollow noopener">61</a><span><span>debconf: falling back to frontend: Readline </span></span>
</div></div>
<div><div>
<a href="https://github.com/aFluxx/Pigero/commit/b949dfd06c0cb21456a7c04b0ad6c3406d7189f4/checks?check_suite_id=292928151#step:5:62" target="_blank" rel="nofollow noopener">62</a><span><span>Processing triggers for php7.3-fpm (7.3.11-1+ubuntu18.04.1+deb.sury.org+1) ... </span></span>
</div></div>
<div><div>
<a href="https://github.com/aFluxx/Pigero/commit/b949dfd06c0cb21456a7c04b0ad6c3406d7189f4/checks?check_suite_id=292928151#step:5:63" target="_blank" rel="nofollow noopener">63</a><span><span>Job for php7.3-fpm.service failed because the control process exited with error code. </span></span>
</div></div>
<div><div>
<a href="https://github.com/aFluxx/Pigero/commit/b949dfd06c0cb21456a7c04b0ad6c3406d7189f4/checks?check_suite_id=292928151#step:5:64" target="_blank" rel="nofollow noopener">64</a><span><span>See "systemctl status php7.3-fpm.service" and "journalctl -xe" for details. </span></span>
</div></div>
<div><div>
<a href="https://github.com/aFluxx/Pigero/commit/b949dfd06c0cb21456a7c04b0ad6c3406d7189f4/checks?check_suite_id=292928151#step:5:65" target="_blank" rel="nofollow noopener">65</a><span><span>invoke-rc.d: initscript php7.3-fpm, action "restart" failed. </span></span>
</div></div>
<div><div>
<a href="https://github.com/aFluxx/Pigero/commit/b949dfd06c0cb21456a7c04b0ad6c3406d7189f4/checks?check_suite_id=292928151#step:5:66" target="_blank" rel="nofollow noopener">66</a><span><span>●</span><span> php7.3-fpm.service - The PHP 7.3 FastCGI Process Manager </span></span>
</div></div>
<div><div>
<a href="https://github.com/aFluxx/Pigero/commit/b949dfd06c0cb21456a7c04b0ad6c3406d7189f4/checks?check_suite_id=292928151#step:5:67" target="_blank" rel="nofollow noopener">67</a><span><span> Loaded: loaded (/lib/systemd/system/php7.3-fpm.service; enabled; vendor preset: enabled) </span></span>
</div></div>
<div><div>
<a href="https://github.com/aFluxx/Pigero/commit/b949dfd06c0cb21456a7c04b0ad6c3406d7189f4/checks?check_suite_id=292928151#step:5:68" target="_blank" rel="nofollow noopener">68</a><span><span> Active: </span><span>failed</span><span> (Result: exit-code) since Sat 2019-11-02 14:25:42 UTC; 10ms ago </span></span>
</div></div>
<div><div>
<a href="https://github.com/aFluxx/Pigero/commit/b949dfd06c0cb21456a7c04b0ad6c3406d7189f4/checks?check_suite_id=292928151#step:5:69" target="_blank" rel="nofollow noopener">69</a><span><span> Docs: man:php-fpm7.3(8) </span></span>
</div></div>
<div><div>
<a href="https://github.com/aFluxx/Pigero/commit/b949dfd06c0cb21456a7c04b0ad6c3406d7189f4/checks?check_suite_id=292928151#step:5:70" target="_blank" rel="nofollow noopener">70</a><span><span> Process: 2846 ExecStart=/usr/sbin/php-fpm7.3 --nodaemonize --fpm-config /etc/php/7.3/fpm/php-fpm.conf </span><span>(code=exited, status=78)</span> </span>
</div></div>
<div><div>
<a href="https://github.com/aFluxx/Pigero/commit/b949dfd06c0cb21456a7c04b0ad6c3406d7189f4/checks?check_suite_id=292928151#step:5:71" target="_blank" rel="nofollow noopener">71</a><span><span> Main PID: 2846 (code=exited, status=78) </span></span>
</div></div>
<div><div><a href="https://github.com/aFluxx/Pigero/commit/b949dfd06c0cb21456a7c04b0ad6c3406d7189f4/checks?check_suite_id=292928151#step:5:72" target="_blank" rel="nofollow noopener">72</a></div></div>
<div><div>
<a href="https://github.com/aFluxx/Pigero/commit/b949dfd06c0cb21456a7c04b0ad6c3406d7189f4/checks?check_suite_id=292928151#step:5:73" target="_blank" rel="nofollow noopener">73</a><span><span>Nov 02 14:25:42 fv-az98 systemd[1]: Starting The PHP 7.3 FastCGI Process Manager... </span></span>
</div></div>
<div><div>
<a href="https://github.com/aFluxx/Pigero/commit/b949dfd06c0cb21456a7c04b0ad6c3406d7189f4/checks?check_suite_id=292928151#step:5:74" target="_blank" rel="nofollow noopener">74</a><span><span>Nov 02 14:25:42 fv-az98 php-fpm7.3[2846]: [02-Nov-2019 14:25:42] ERROR: unable to bind listening socket for address '/run/php/php7.3-fpm.sock': No such file or directory (2) </span></span>
</div></div>
<div><div>
<a href="https://github.com/aFluxx/Pigero/commit/b949dfd06c0cb21456a7c04b0ad6c3406d7189f4/checks?check_suite_id=292928151#step:5:75" target="_blank" rel="nofollow noopener">75</a><span><span>Nov 02 14:25:42 fv-az98 php-fpm7.3[2846]: [02-Nov-2019 14:25:42] ERROR: FPM initialization failed </span></span>
</div></div>
<div><div>
<a href="https://github.com/aFluxx/Pigero/commit/b949dfd06c0cb21456a7c04b0ad6c3406d7189f4/checks?check_suite_id=292928151#step:5:76" target="_blank" rel="nofollow noopener">76</a><span><span>Nov 02 14:25:42 fv-az98 systemd[1]: </span><span>php7.3-fpm.service: Main process exited, code=exited, status=78/n/a</span> </span>
</div></div>
<div><div>
<a href="https://github.com/aFluxx/Pigero/commit/b949dfd06c0cb21456a7c04b0ad6c3406d7189f4/checks?check_suite_id=292928151#step:5:77" target="_blank" rel="nofollow noopener">77</a><span><span>Nov 02 14:25:42 fv-az98 systemd[1]: </span><span>php7.3-fpm.service: Failed with result 'exit-code'.</span> </span>
</div></div>
<div><div>
<a href="https://github.com/aFluxx/Pigero/commit/b949dfd06c0cb21456a7c04b0ad6c3406d7189f4/checks?check_suite_id=292928151#step:5:78" target="_blank" rel="nofollow noopener">78</a><span><span>Nov 02 14:25:42 fv-az98 systemd[1]: </span><span>Failed to start The PHP 7.3 FastCGI Process Manager.</span> </span>
</div></div>
<div><div>
<a href="https://github.com/aFluxx/Pigero/commit/b949dfd06c0cb21456a7c04b0ad6c3406d7189f4/checks?check_suite_id=292928151#step:5:79" target="_blank" rel="nofollow noopener">79</a><span><span>dpkg: error processing package php7.3-fpm (--configure): </span></span>
</div></div>
<div><div>
<a href="https://github.com/aFluxx/Pigero/commit/b949dfd06c0cb21456a7c04b0ad6c3406d7189f4/checks?check_suite_id=292928151#step:5:80" target="_blank" rel="nofollow noopener">80</a><span><span> installed php7.3-fpm package post-installation script subprocess returned error exit status 1 </span></span>
</div></div>
<div><div>
<a href="https://github.com/aFluxx/Pigero/commit/b949dfd06c0cb21456a7c04b0ad6c3406d7189f4/checks?check_suite_id=292928151#step:5:81" target="_blank" rel="nofollow noopener">81</a><span><span>E: Sub-process /usr/bin/dpkg returned an error code (1) </span></span>
</div></div>
<div><div>
<a href="https://github.com/aFluxx/Pigero/commit/b949dfd06c0cb21456a7c04b0ad6c3406d7189f4/checks?check_suite_id=292928151#step:5:82" target="_blank" rel="nofollow noopener">82</a><span><span>##[error]Process completed with exit code 100.</span></span>
</div></div>Does anyone know how I can install php73-intl without the FastCGI Process Manager to fail. One of my composer dependecies requires the intl extension, so I can't run
<pre><code>composer install
</code></pre>
<p>without it</p>
<p>Any suggestions?</p>
<p>Kind Regards,<br>
Arne</p>

**Comment by shivammathur** [comment, marked as answer]
<p>You can use this GitHub Action to install ext-intl.<br>
<a href="https://github.com/shivammathur/setup-php" rel="nofollow noopener">https://github.com/shivammathur/setup-php</a></p>
<pre><code>name: Setup PHP
uses: shivammathur/setup-php@v2
with:
  php-version: '7.4'
  extensions: intl
name: Composer Install
run: composer install
</code></pre>

**Comment by mhitza** [comment]
<p>Thank you for the setup-php action. You should update your comment to use ‘extensions’ instead of ‘extension-csv’ as you’re printing a deprecation message right now.</p>

**Comment by DeBelserArne** [comment]
<p>Hello there,</p>
<p>I’ve recently came across a .yml which used </p>
<pre><code>uses: shivammathur/setup-php@v1
</code></pre>
<p>Right now I’m running it like:</p>
<pre><code>services:
      redis:
        image: redis
        ports:
            - 6379:6379
        options: --entrypoint redis-server
    steps:
    - uses: actions/checkout@v1

    - name: Setup PHP
      uses: shivammathur/setup-php@v1
      with:
        php-version: '7.4'
        extensions: dom, curl, libxml, mbstring, zip, pcntl, pdo, sqlite, pdo_sqlite, bcmath, soap, intl, gd, exif, iconv, imagick
        coverage: none
</code></pre>
<p>Also, I ran into problems with Redis, on my own machine I had it installed, but the tests which triggerd it would fail. In my new yml configuration I was able to tackle this. (See above)</p>
<p>Thank you all for helping me out!</p>

**Comment by shivammathur** [comment]
<p>Updated the answer.</p>
